### ISSUE TITLE
Background toys with negative test statistic

### ISSUE THREAD

**Issue by fionnbishop** [gammacombo/gammacombo#213]
I am finding that background toys with negative test statistic (b_teststat_toy) can enter into the CLs calculation, when I believe these ought to be rejected. I believe this could be corrected by adding &&b_teststat_toy>=0. in line 613, but I can't be completely sure that (a) this interpretation is correct and (b) making this change wouldn't have unintended effects.

https://github.com/gammacombo/gammacombo/blob/3d43b971a211ade23e83a2612dd73ec516e37362/core/src/MethodDatasetsPluginScan.cpp#L613-L621

**Comment by tmombaecher** [comment]
We normally don't reject those, as we don't for the non-background toys. It's better to do this consistently. 
Negative test statistics should be few and comparable to teststat=0.
However, if you observe significant negative fluctuations, this would indicate some trouble with your fit setup and one should indeed reject those. Can you give some more detail on how negative those get?

**Comment by fionnbishop** [comment]
Are they not rejected for S+B toys here?
https://github.com/gammacombo/gammacombo/blob/3d43b971a211ade23e83a2612dd73ec516e37362/core/src/MethodDatasetsPluginScan.cpp#L603-L606

Having looked properly, there are a lot more such toys than I expected and they become quite negative, so there is probably a problem on my end.

**Comment by tmombaecher** [comment]
You are right, the treatment is not perfectly equal here, which can maybe have an effect the results in the case of a significant fraction of negative test statistics. I made a quick change in my fork.
However, many and largely negative test statistic values are indeedan indicator for some trouble in the fit. Where did you spot them? We should flag this issue with a control plot.
It is unfortunately not trivial on how to proceed if you cannot avoid those.
I know the CMS-LHCb combination did deal with sucha problem, but I still don't understand how.

**Comment by fionnbishop** [comment]
Thanks for making this change.
It would indeed be very helpful to have GammaCombo flag up toys which have been rejected due to negative test stat - I would find it very helpful to at least have the % rejected printed alongside the % failed toys.
I think I have realised which feature of the workspaces I pass to GammaCombo is causing this problem, which should be easy enough to change.

**Comment by tmombaecher** [comment]
I had a little closer look. Actually we had already a plot (in the `debug_plots_{scanVar}.) and a print out ("unphysical toys" or "background toys") for the s+b toys. But I modified it a little to print and plot also the negative background test statistic rates (and clearly stating which are the negative test stat rates).
If you have a strong disagreement between the neg test stat toys for background and the neg test stat for the plugin toys (at low signal) there might be an issue with you background model.
This can to be sometimes related to specifying an explicit background model in 
https://github.com/gammacombo/gammacombo/blob/3d43b971a211ade23e83a2612dd73ec516e37362/tutorial/main/tutorial_dataset.cpp#L71
(maybe also due to ROOT-internal reasons). If you comment out this line, the tool will define the background model as the s+b model with `scanVar=0`, which seems to be more stable. I added a comment in the tutorial regarding this.
